.clang-tidy

```diff
@@ -1,42 +1,32 @@
 Checks: "-*,\
 bugprone-*,\
+cert-dcl21-cpp,\
+cert-dcl50-cpp,\
+cert-env33-c,\
+-cert-err34-c,\
+cert-err52-cpp,\
+-cert-err58-cpp,\
+cert-err60-cpp,\
+-cert-flp30-c,\
 -clang-analyzer-security.FloatLoopCounter,\
 google-build-namespaces,\
--google-explicit-constructor,\
+google-explicit-constructor,\
```

Comment:
I have mixed feelings about this. From a purity point of view, I agree and ideally it would have been enabled from day 1. However, this PR introduces a lot of `// NOLINT` Comments to combat the difficult cases. I think this is probably worse from a readability point of view

Reply:
Some types are only difficult cases because there is widespread reliance on implicit construction for these types in the code. Enabling this check and shooting down further NOLINTs in code review will at least let us keep the problem in check.

As an aside, not all the NOLINTs are in place for constructors which would be too tedious to make explicit: I allowed implicit conversions between `vm::Ptr<Base>` and `vm::Ptr<Derived>`, for example, as this is reasonable and mirrors stdlib smart pointer behaviour. Construction of `vm::Ptr<T>` from `T*`, on the other hand is made explicit.